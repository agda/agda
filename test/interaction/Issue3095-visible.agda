open import Agda.Builtin.Nat

foo : Nat → Nat
foo x = bar x
  where
    bar : Nat → Nat
    bar y = {!x!}

-- WAS: Splitting on x produces the following garbage:
-- foo : Nat → Nat
-- foo x = bar x
--   where
--     bar : Nat → Nat
--     bar y = {!!}
--     bar y = {!!}

-- SHOULD: raise an error
