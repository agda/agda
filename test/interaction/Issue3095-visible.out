(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Nat " nil)
((last . 1) . (agda2-goals-action '(0)))
(agda2-info-action "*Error*" "Issue3095-visible.agda:7,13-18 Cannot split on module parameter x when checking that the expression ? has type Nat" nil)
((last . 3) . (agda2-maybe-goto '("Issue3095-visible.agda" . 101)))
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
